src/main.rs

```diff
@@ -20,45 +20,36 @@ use addr2line::fallible_iterator::FallibleIterator as _;
 use anyhow::{anyhow, bail, Context};
 use arrayref::array_ref;
 use colored::Colorize as _;
+use defmt_decoder::DEFMT_VERSION;
 use gimli::{
-    read::{CfaRule, DebugFrame, UnwindSection},
-    BaseAddresses, EndianSlice, LittleEndian, RegisterRule, UninitializedUnwindContext,
+    read::{DebugFrame, UnwindSection},
+    BaseAddresses, LittleEndian, UninitializedUnwindContext,
 };
 use log::Level;
 use object::{
     read::{File as ElfFile, Object as _, ObjectSection as _},
     ObjectSegment, ObjectSymbol, SymbolSection,
 };
-use probe_rs::config::{registry, MemoryRegion, RamRegion};
 use probe_rs::{
+    config::{registry, MemoryRegion, RamRegion},
     flashing::{self, Format},
-    Core, CoreRegisterAddress, DebugProbeInfo, MemoryInterface, Probe, Session,
+    Core, DebugProbeInfo, MemoryInterface, Probe, Session,
 };
 use probe_rs_rtt::{Rtt, ScanRegion, UpChannel};
 use signal_hook::consts::signal;
 use structopt::{clap::AppSettings, StructOpt};
 
-const TIMEOUT: Duration = Duration::from_secs(1);
+use crate::{
+    registers::{Registers, LR, LR_END, PC, SP},
+    stacked::Stacked,
+};
+
 const STACK_CANARY: u8 = 0xAA;
+const SIGABRT: i32 = 134;
+/// Successfull termination of process.
+const SIGSUCCESS: i32 = 0;
```

Comment:
Success is not a signal, so this should not start with `SIG`. libc calls this `EXIT_SUCCESS`.

Reply:
Okay. I was looking into linuxs signals for 15 minutes, but didn't really find a proper solution, so I just made up this term 😓